src/registrar/models/dns/dns_account.py

```diff
@@ -9,7 +13,12 @@ class DnsAccount(TimeStampedModel):
     )  # type: ignore
 
     @property
-    def get_x_account_id_by_name_if_exists(self):
-        link = self.account_link.filter(is_active=True).select_related("vendor_dns_account").first()
+    def x_account_id(self):
+        try:
+            x_account_id = self.account_link.get(is_active=True).vendor_dns_account.x_account_id
```

Comment:
I don't know if this is more efficient or not, but it does expose if there is no active account since the `get` would throw an error if no account was `active`.